src/WinRT.Runtime/ExceptionHelpers.cs

```diff
@@ -99,15 +118,12 @@ private static Exception GetExceptionForHR(int hr, bool useGlobalErrorState, out
             string restrictedCapabilitySid = null;
             bool hasOtherLanguageException = false;
 
-            if (useGlobalErrorState && getRestrictedErrorInfo != null)
+            if (useGlobalErrorState)
             {
-                Marshal.ThrowExceptionForHR(getRestrictedErrorInfo(out IntPtr restrictedErrorInfoPtr));
-
-                if (restrictedErrorInfoPtr != IntPtr.Zero)
-                {
-                    IObjectReference restrictedErrorInfoRef = ObjectReference<ABI.WinRT.Interop.IRestrictedErrorInfo.Vftbl>.Attach(ref restrictedErrorInfoPtr);
+                var restrictedErrorInfoRef = BorrowRestrictedErrorInfo();
```

Comment:
consider **_using_** var restrictedErrorInfoRef = .. here and below to make it do the cleanup here rather than during GC thread.